components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4557,4 +4561,36 @@ public static boolean isFederatedRoleBasedAuthzEnabled(String clientId) throws I
         }
         return isFederatedRoleBasedAuthzEnabled;
     }
+
+    /**
+     * Is carbon role based validation enabled for first level organizations in the deployment.
+     *
+     * @return True if carbon role based validation enabled for first level organizations.
+     */
+    public static boolean isCarbonRoleValidationEnabledForLevelOneOrgs() {
+
+        return Boolean.parseBoolean(IdentityUtil.getProperty(IS_CARBON_ROLE_VALIDATION_ENABLED_FOR_LEVEL_ONE_ORGS));
+    }
+
+    /**
+     * Return whether organization role based validation is used.
+     *
+     * @param organizationId Organization id.
+     * @return False if the organization is a first level organization in the deployment and
+     * IS_CARBON_ROLE_VALIDATION_ENABLED_FOR_LEVEL_ONE_ORGS config is enabled. Other true.
+     */
+    public static boolean useOrganizationRolesForValidation(String organizationId) {
+
+        if (!isCarbonRoleValidationEnabledForLevelOneOrgs()) {
+            return true;
+        }
+        try {
+            // Return false if the organization is in depth 1.
+            return OAuth2ServiceComponentHolder.getOrganizationManagementService()
+                    .getOrganizationDepthInHierarchy(organizationId) != 1;
+        } catch (OrganizationManagementServerException e) {
+            log.error("Error while checking the depth of the given organization.");
+        }
+        return true;
```

Comment:
In case of exception occured.
```suggestion
        return false;
```